lib/zlib.js

```diff
@@ -445,9 +445,11 @@ Zlib.prototype.flush = function(kind, callback) {
       this.once('end', callback);
   } else if (ws.needDrain) {
     var self = this;
-    this.once('drain', function() {
-      self.flush(callback);
-    });
+    if (callback) {
+      this.once('drain', function() {
+        self.flush(callback);
+      });
```

Comment:
I think I spot a bug here: the call should be `self.flush(kind, callback)`, it forgets the `kind` now.

Also, I'd ES6 this up and write it as `this.once('drain', () => this.flush(kind, callback))`.